api/v1beta1/cluster_types.go

```diff
@@ -878,6 +885,7 @@ type ClusterNetwork struct {
 // NetworkRanges represents ranges of network addresses.
 type NetworkRanges struct {
 	// cidrBlocks is a list of CIDR blocks.
+	// +kubebuilder:validation:MaxItems=1000
```

Comment:
```suggestion
	// +kubebuilder:validation:MaxItems=100
```
Should be good as well IMO